python/dgl/dataloading/pytorch/__init__.py

```diff
@@ -242,9 +297,25 @@ def __init__(self, g, nids, block_sampler, device='cpu', **kwargs):
             self.is_distributed = True
         else:
             self.collator = _NodeCollator(g, nids, block_sampler, **collator_kwargs)
-            self.dataloader = DataLoader(self.collator.dataset,
-                                         collate_fn=self.collator.collate,
-                                         **dataloader_kwargs)
+            if "num_workers" in dataloader_kwargs and dataloader_kwargs["num_workers"] > 0:
+                assert g.device == th.device('cpu'), \
+                    'Only graphs on the cpu are supported for NodeDataLoader ' \
+                    'in the case of num_workers > 0.'
+                self.dataloader = DataLoader(
+                    self.collator.dataset,
+                    collate_fn=self.collator.collate,
+                    **dataloader_kwargs)
+            else:
+                if isinstance(self.collator.dataset, th.Tensor):
```

Comment:
if this is more efficient, how about we convert `dataset` to `th.Tensor`. so users don't see weird performance difference.

Reply:
In 
d7c8619 if it's a DGL tensor, it will get converted to PyTorch tensor.